### ISSUE TITLE
Handle dns change

### ISSUE THREAD

**Issue by NoUseFreak** [domnikl/statsd-php#67]
Hi,

First of, thank for this great library!

I am running into a problem when our statsd server changes DNS, the library does not seem to re-resolve the DNS record resulting in a loss of metrics. 
I am running a statsd server (datadog) in kubernetes, when the statsd node gets rescheduled to a different node, the DNS record changes. We are connecting to it over UDP and are not using a persistent connection.

Do you have any way to resolve this issue?

**Comment by domnikl** [comment]
Hi thanks, I am glad you found it useful! 
Currently, the UDPSocket will only try to connect once, but will never try to reconnect: https://github.com/domnikl/statsd-php/blob/master/src/Connection/UdpSocket.php#L80
I would not change that existing behaviour but the class could be configured to try to reconnect.

**Comment by NoUseFreak** [comment]
I've been doing some investigation, do you think the following would work?

```
    protected function isConnected(): bool
    {
        return (feof($this->socket) === true);
    }
```

**Comment by domnikl** [comment]
Looks good, except `$this->socket` may be also be null if there was a connection error or it just hasn't connected because no messages have been sent yet. I can prepare a PR which you can test against if you like.

**Comment by domnikl** [comment]
merged and tagged 3.0.1
